mobile/lib/providers/asset.provider.dart

```diff
@@ -78,27 +80,48 @@ class AssetNotifier extends StateNotifier<bool> {
   }
 
   Future<bool> deleteLocalOnlyAssets(
+    BuildContext context,
     Iterable<Asset> deleteAssets, {
     bool onlyBackedUp = false,
   }) async {
     _deleteInProgress = true;
     state = true;
     try {
+      // Filter local assets based on the backup status (`isRemote`).
+      // If `onlyBackedUp` is true, only select assets that are also present on the server (`isRemote`).
+      final assetsToDelete = deleteAssets
+          .where((e) => e.isLocal && (!onlyBackedUp || e.isRemote))
+          .toList();
+
+      if (assetsToDelete.isEmpty) {
+        // No assets to delete, exit early
+        ImmichToast.show(
+          context: context,
+          msg: 'No backed-up assets selected for deletion.',
+          gravity: ToastGravity.BOTTOM,
+        );
+        return false;
+      }
+
+      // Continue with deletion of the filtered assets
       final assets = onlyBackedUp
-          ? deleteAssets.where((e) => e.storage == AssetState.merged)
-          : deleteAssets;
+          ? assetsToDelete.where((e) => e.storage == AssetState.merged)
+          : assetsToDelete;
+
       final localDeleted = await _deleteLocalAssets(assets);
       if (localDeleted.isNotEmpty) {
-        final localOnlyIds = deleteAssets
+        final localOnlyIds = assetsToDelete
             .where((e) => e.storage == AssetState.local)
             .map((e) => e.id)
             .toList();
         // Update merged assets to remote only
-        final mergedAssets =
-            deleteAssets.where((e) => e.storage == AssetState.merged).map((e) {
```

Comment:
This portion of the code is actually where the logical error is. So if we do not need the toast message if there are not assets to delete, the entirety of the change would only be switching the list from `deleteAssets` to `assets` in the current implementation

Basically, the issue is that, we filter the list of assets and get the final list to delete but when making the changes to the local DB, we filter the assets again from the original list passed and not the filtered list